### ISSUE TITLE
Please add administrator user as optional parameter to bin/magento setup:upgrade

### ISSUE THREAD

**Issue by nagno** [magento/magento2#1671]
Hello!

We tried to create a cms page from setup script and it seems, there is no clean way to do it with the merchant beta version.
The issue is caused by the observer in the VersionsCms module. The new CMS page is always going to be handled like a version controlled CMS page (https://github.com/magento/magento2ee/blob/f37c9d7bea3ff2300dc38f2e426f0f447b6c4c4d/app/code/Magento/VersionsCms/Model/Backend/Observer.php#L274) and it requires a logged in admin user (https://github.com/magento/magento2ee/blob/f37c9d7bea3ff2300dc38f2e426f0f447b6c4c4d/app/code/Magento/VersionsCms/Model/Backend/Observer.php#L291)

It would be a lot easier to implement these kind of setup scripts, if we have an admin user loaded by default.

Could you please consider to add the optional parameter for the administrator username like the sample data installer works at the moment?

Thanks!

**Comment by mbrinton01** [comment]
@nagno it is possible to specify an admin username during CLI install: http://devdocs.magento.com/guides/v2.0/install-gde/install/cli/install-cli-install.html

Is this what you are referring to?

**Comment by nagno** [comment]
Hi @mbrinton01,
nope, I'm referring to the bin/magento setup:upgrade command (as it is in the title of the issue).
Unfortunately it doesn't have the admin user option:
```bash
bin/magento help setup:upgrade
Usage:
 setup:upgrade [--magento-init-params="..."]

Options:
 --magento-init-params  Add to any command to customize Magento initialization parameters
                        For example: "MAGE_MODE=developer&MAGE_DIRS[base][path]=/var/www/example.com&MAGE_DIRS[cache][path]=/var/tmp/cache"
 --help (-h)            Display this help message
 --quiet (-q)           Do not output any message
 --verbose (-v|vv|vvv)  Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug
 --version (-V)         Display this application version
 --ansi                 Force ANSI output
 --no-ansi              Disable ANSI output
 --no-interaction (-n)  Do not ask any interactive question
```

**Comment by mazhalai** [comment]
@nagno Thank you for reporting this issue, we have created MAGETWO-41500 to track this internally.

**Comment by nagno** [comment]
Thanks!

**Comment by BaDos** [comment]
Hello, @nagno!

How did you create a CMS page?
You can see the sample code to create page in the next file: https://github.com/magento/magento2/blob/develop/app/code/Magento/Cms/Setup/InstallData.php

I created simple module and added creation of CMS page in setup script.
Ran bin/magento setup:upgrade
New module was installed and new CMS page was created.

**Comment by nagno** [comment]
Hi @BaDos,
I would like to highlight, we are using the merchant beta version (beta 3), not the simple beta version. Have you tried with that?
The simple beta version doesn't have the VersionCms module, which causes the issue.

We have a code like this:
```php
$page = $this->pageFactory->create()->load($pageData['identifier'], 'identifier');
if (!$page->getId()) {
    $page->setIdentifier($pageData['identifier']);
}
$page->setTitle($pageData['title']);
$page->setPageLayout($pageData['page_layout']);
$page->setMetaKeywords($pageData['meta_keywords']);
$page->setMetaDescription($pageData['meta_description']);
$page->setContentHeading($pageData['content_heading']);
$page->setContent($pageData['content']);
$page->setIsActive($pageData['is_active']);
$page->setStores($pageData['stores']);
$page->setSortOrder($pageData['sort_order']);
$page->save();
```

**Comment by nagno** [comment]
I think, this issue was reported before we started to log the issues to the partner portal. Do you want me to report it on the partner portal instead of github?

**Comment by piotrekkaminski** [comment]
@nagno do you still have this issue in the latest version?

**Comment by nagno** [comment]
Hi @piotrekkaminski,
Could you please let me know, why did you close the ticket?

**Comment by adragus-inviqa** [comment]
@piotrekkaminski - I can confirm the issue on the merchant beta 5 version.
Can you guys re-open this? 
You cannot add pages from setup scripts with the `VersionCms` enabled.

**Comment by nagno** [comment]
I opened a partner portal ticket (#02570269), probably that will help to get support on it.

@piotrekkaminski I'm really sad it was closed without any explanation. :(
I know the Magento developers are really busy during these days (so am I) but closing tickets without really investigating them is not a good idea. For everybody's sake please don't continue this practice or we will end up with a product full of bugs.

**Comment by jamescowie** [comment]
Re opened the issue for more clarification. @nagno @adragus-inviqa can you confirm if you are still having issues ? @mazhalai Can you share the status of the internal ticket please ?

**Comment by nagno** [comment]
I tried to reproduce it one time, but I wasn't successful with a single setup script, however I'm quite sure the issue is there. I think, it is happening when you have multiple scripts and they somehow causes this issue. Also it might be from upgrade script. We had the issue on our project, where we had many setup scripts.
I have to spend a little bit more time on it to find a clear instruction set to reproduce it, but recently I was too busy and I will be in the next 2-3 weeks. @adragus-inviqa: if you have some time, I'm happy if you can add the steps to reproduce it on a vanilla environment.

**Comment by mazhalai** [comment]
@nagno any update on this?

**Comment by nagno** [comment]
Hi @mazhalai 
I was heavily working on it in the last 2 days and finally I just successfully reproduced it (after ~4.5 months it is not easy to roll back a project to a specific state to reproduce an issue). I'm quite sure it is related to a previous setup script state, but unfortunately I just ran out of my free time again and I'm won't have time investigate it further in the next few weeks. :( I'm just about to send the entire project code to your support team via partner portal with the instructions to reproduce it.

-- UPDATE --
I tested it with merchant_beta 9.
